src/modules/ekf2/EKF2.cpp

```diff
@@ -843,6 +843,56 @@ void EKF2::PublishLocalPosition(const hrt_abstime &timestamp)
 	_local_position_pub.publish(lpos);
 }
 
+matrix::SquareMatrix<float, 3> EKF2::propagate_covariances_from_quat_to_euler(const matrix::Quatf &q,
+		const matrix::SquareMatrix<float, 4> &quat_covariances) const
+{
+	// Jacobian matrix (3x4) containing the partial derivatives of the
+	// Euler angle equations with respect to the quaternions
+	matrix::Matrix<float, 3, 4> G;
+
+	// quaternion components
+	float q1 = q(0);
+	float q2 = q(1);
+	float q3 = q(2);
+	float q4 = q(3);
+
+	// numerator components
+	float n1 =  2 * q1 * q2 + 2 * q2 * q4;
```

Comment:
Where did these come from?

Reply:
I will see if this derivation can be added to our existing sympy script.

Reply:
See https://github.com/priseborough/PX4-Autopilot/tree/pr-ekf2_derivation_update for a symbolic derivation

Reply:
Nuno's original PR for context. https://github.com/PX4/PX4-Autopilot/pull/11092